docs/source/howto/codes.rst

```diff
@@ -45,8 +45,84 @@ Design guidelines
 Running external codes
 ======================
 
-`#3987`_
+To run an external code with AiiDA, you will need to use an appropriate :ref:`calculation plugin <topics:plugins>` that knows how to take transform the input nodes into the input files that the code expects, copy everything in the code's machine, run the calculation and retrieve the results.
+This plugin would have already been created, either by you (see section on :ref:`interfacing external codes <how-to:codes:plugin>`) or by other AiiDA plugin-developers (available packages are listed in the `plugin registry <https://aiidateam.github.io/aiida-registry/>`_), and has been properly installed in your environment.
+You can check which ones you have currently available by running the ``verdi plugin`` command:
 
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations
+
+For the purpose of this example, we will use the ``arithmetic.add`` plugin, which is a plugin for bash to sum two integers that comes pre-installed with ``aiida-core``.
+You can access it through the use of the ``CalculationFactory``:
+
+.. code-block:: python
+
+    from aiida.plugins import CalculationFactory
+    calculation_class = CalculationFactory('arithmetic.add')
+
+We will obviously also need to provide the inputs for the code when running the calculation.
+You can also use ``verdi plugin`` to check this and other relevant information for specific plugins:
+
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations arithmetic.add
```

Comment:
Please add the output here as well. We don't expect users to run these commands to be able to follow the example.

Reply:
Mmm, but from what I understood from @sphuber 's [previous review](https://github.com/aiidateam/aiida-core/pull/4114#discussion_r430023980), the idea was to not include the actual information in order for it to be easier to maintain. Maybe I misunderstood that?

Reply:
That's not how I understand that comment and I was missing the output when I read through this how-to.

Reply:
The idea was indeed to not have a copy of the spec here in case the implementation changes and the documentation goes out of date. Pasting the output of `verdi plugin list` has the same downside, but I can see why it is important to show it, so fine for me to add it. Maybe don't show the exit codes as they are not necessary and just do description, inputs and outputs.

Anyway, using this over copying source code is still better as they can then execute the command themselves to try and they can apply it to any other plugin, for example the one they would actually like to launch.